From 03e2fc6061e66e3a0b6148805fe70894f6bc255f Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 22 Mar 2024 16:16:48 +0100
Subject: [PATCH 0829/1695] plat-stm32mp1: STM32MP15 without secure service can
 suspend without RNG

STM32MP15 variant with CFG_STM32MP1_OPTEE_IN_SYSRAM=y requires
STM32 RNG driver to support PM suspend/resume sequences because
core executes from internal secure SRAM that needs to be ciphered
(with Authenticated Encryption) and saved to external which process
requires RNG support.

By the way, replace the #ifndef directive on STM32MP13 variant with
a if(IS_ENABLED()) condition on STM32MP15 variant.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 64a1a14fb86831753003d9e251acc10c1bf8ba5b plat-stm32mp1: STM32MP15 without secure service can suspend without RNG]
Change-Id: I081f01287e6bc0f15d5baaabc554f0e36a273565
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/376014
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/pm/psci.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp1/pm/psci.c b/core/arch/arm/plat-stm32mp1/pm/psci.c
index 38513fd88..45a8bca9c 100644
--- a/core/arch/arm/plat-stm32mp1/pm/psci.c
+++ b/core/arch/arm/plat-stm32mp1/pm/psci.c
@@ -302,10 +302,10 @@ static bool plat_can_suspend(void)
 	uint32_t exceptions = 0;
 	bool rc = true;
 
-#ifndef CFG_STM32MP13
-	if (!IS_ENABLED(CFG_STM32_RNG))
+	/* STM32MP15 in SYSRAM requires STM32 RNG in order to suspend */
+	if (IS_ENABLED(CFG_STM32MP15) && !IS_ENABLED(CFG_STM32_RNG) &&
+	    IS_ENABLED(CFG_STM32MP1_OPTEE_IN_SYSRAM))
 		return false;
-#endif
 
 	if (CFG_TEE_CORE_NB_CORE == 1)
 		return true;
-- 
2.47.2

